s96at/CMakeLists.txt

```diff
@@ -27,12 +27,18 @@ set(I2C_DEVICE "/dev/i2c-0")
 add_definitions(-DCMAKE_BUILD_TYPE=Debug)
 
 add_library(${PROJECT_NAME} SHARED ${SRC})
-set_target_properties(${PROJECT_NAME} PROPERTIES PUBLIC_HEADER include/s96at.h)
+
+set(PUBLIC_HEADERS ${CMAKE_SOURCE_DIR}/include/s96at.h
+		   ${CMAKE_SOURCE_DIR}/include/s96at_private.h)
+set_target_properties(${PROJECT_NAME} PROPERTIES PUBLIC_HEADER "${PUBLIC_HEADERS}")
 
 target_compile_definitions(${PROJECT_NAME}
 	PRIVATE -DI2C_DEVICE="${I2C_DEVICE}"
 	PRIVATE -DDEBUG
 )
+set(CMAKE_INSTALL_PREFIX /usr)
```

Comment:
I think it would be better to use something like this (might be that syntax needs some tweak).
```Makefile
include(GNUInstallDirs)
...

install (TARGETS ${PROJECT_NAME} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
     PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
```

By doing this, things will end up in pre-defined GNU directories. What do you think about this?

See this for more information about this CMake directive: https://cmake.org/cmake/help/v3.0/module/GNUInstallDirs.html

Reply:
I wasn't aware of this directive, thanks for pointing it out. I updated the patch according to your suggestion, with the only exception of installing headers under 
```
${CMAKE_INSTALL_INCLUDEDIR}/secure96 
```
so that we can later add the headers for the TPM module under that dir too.

Reply:
By the way, the `set(CMAKE_INSTALL_PREFIX /usr)` line isn't that superfluous now?

Reply:
Okay, so the truth is that I forgot to remove it. Yet, the default behaviour of GNUInstallDirs is to install everything under /usr/local.  The GNU documentation (https://www.gnu.org/prep/standards/html_node/Directory-Variables.html) states:
>‘includedir’
>
>    The directory for installing header files to be included by user programs with the C ‘#include’ preprocessor directive. This should normally be /usr/local/include, but write it as $(prefix)/include. (If you are using Autoconf, write it as ‘@includedir@’.)

The default path looks a bit odd to me. In my system there is nothing under /usr/local/include. On the other hand, the Linux Filesystem Hierarchy Standard (https://refspecs.linuxfoundation.org/FHS_3.0/fhs-3.0.pdf) defines /usr/local as 
> The /usr/local hierarchy is for use by the system administrator when installing software locally. It needs to be safe from being overwritten when the system software is updated. It may be used for programs and data that are shareable amongst a group of hosts, but not found in /usr.
Locally  installed  software  must  be  placed  within  /usr/local rather  than  /usr unless  it  is  being installed to replace or upgrade software in /usr. 

Not sure how to interpret this. To me it sounds like if you use "make install" then files should go to /usr/local, but if you create distribution packages one should override the prefix to /usr?

Reply:
@jbech-linaro no problem, removed the fixed prefix.